deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxListService.java

```diff
@@ -0,0 +1,325 @@
+import ch.cyberduck.core.deepbox.io.swagger.client.model.Box;
+import ch.cyberduck.core.deepbox.io.swagger.client.model.Boxes;
+import ch.cyberduck.core.deepbox.io.swagger.client.model.DeepBox;
+import ch.cyberduck.core.deepbox.io.swagger.client.model.DeepBoxes;
+import ch.cyberduck.core.deepbox.io.swagger.client.model.Node;
+import ch.cyberduck.core.deepbox.io.swagger.client.model.NodeContent;
+import ch.cyberduck.core.exception.AccessDeniedException;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.exception.ConnectionCanceledException;
+import ch.cyberduck.core.preferences.HostPreferences;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.text.MessageFormat;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import static ch.cyberduck.core.deepbox.DeepboxAttributesFinderFeature.CANLISTCHILDREN;
+
+public class DeepboxListService implements ListService {
+    private static final Logger log = LogManager.getLogger(DeepboxListService.class);
+
+    private final DeepboxSession session;
+    private final DeepboxIdProvider fileid;
+    private final DeepboxAttributesFinderFeature attributes;
+    private final int chunksize;
+    private final DeepboxPathContainerService containerService = new DeepboxPathContainerService();
+
+    public DeepboxListService(final DeepboxSession session, final DeepboxIdProvider fileid) {
+        this.session = session;
+        this.fileid = fileid;
+        this.attributes = new DeepboxAttributesFinderFeature(session, fileid);
+        this.chunksize = new HostPreferences(session.getHost()).getInteger("deepbox.listing.chunksize");
+    }
+
+    @Override
+    public AttributedList<Path> list(final Path directory, final ListProgressListener listener) throws BackgroundException {
+        final AttributedList<Path> list = new AttributedList<>();
+        final HashSet<String> closed = new HashSet<>();
+        try {
+            if(directory.isRoot()) {
+                this.listDeepBoxes(directory, listener, list);
+            }
+            else {
+                final String deepBoxNodeId = fileid.getDeepBoxNodeId(directory);
+                if(containerService.isDeepbox(directory)) { // in DeepBox
+                    this.listBoxes(directory, listener, list);
+                }
+                else {
+                    final String boxNodeId = fileid.getBoxNodeId(directory);
+                    if(containerService.isBox(directory)) { // in Box
+                        this.listBox(directory, listener, deepBoxNodeId, boxNodeId, list);
+                    }
+                    else if(containerService.isThirdLevel(directory)) { // in Inbox/Documents/Trash
+                        // N.B. although Documents and Trash have a nodeId, calling the listFiles1/listTrash1 API with parentNode fails!
```

Comment:
Does that imply that attempting to find using `DefaultFindFeature` in _Documents_ will fail?